### PR TITLE
Fix unit tests

### PR DESCRIPTION
**Pull request by bardkw**
Fix unit tests for SMQTK-Image-IO breakout

### CODE REVIEW

**Comment by Purg** [comment]
@bardkw With the merger of the restructure branch, please rebase this.

**Comment by Purg** [comment]
Either between this PR or the other (which looks like it adds mypy fixes?), please add the github actions workflow for CI.

**Comment by Purg** [comment]
Nevermind, just created #6

**Comment by Purg** [comment]
`tests/data/test_file.dat` seems to not have its git history past your restructure.

File: smqtk_image_io/impls/image_reader/gdal_io.py [hunk 1]
```diff
@@ -406,8 +405,9 @@ def _load_as_matrix(self, data_element, pixel_crop=None):
 
         """
         if data_element.is_empty():
-            raise ValueError("{} cannot load 0-sized data (no bytes in {})."
-                             .format(self.name, data_element))
+            raise ValueError(
+                "GdalImageReader cannot load 0-sized data (no bytes in {})."
```
[the remark this answers is not in the document]
Comment (bardkw) [reply smqtk_image_io/impls/image_reader/gdal_io.py:409]: `GdalImageReader` didnt have an explicit `.name` attribute and this was causing an error in pytest

File: tests/utils/test_image_utils.py [hunk 2]
```diff
@@ -67,6 +68,7 @@ def test_invalid_image_returns_false(self):
         assert is_valid_element(self.non_image, check_image=True) is False
 
 
+@pytest.mark.skip(reason="need to replace check images from SMQTK/bin")
```
[the remark this answers is not in the document]
Comment (bardkw) [reply tests/utils/test_image_utils.py:71]: that was my initial thought, but I wasn't sure and marked it as skipped instead. I can go ahead and remove this

File: tests/test_bounding_box.py [hunk 3]
```diff
@@ -98,8 +98,8 @@ def test_bbox_set_vertices_maintain_type_float():
     AxisAlignedBoundingBox._set_vertices(m_bb, minv, maxv)
 
     # BOTH vertices should be integer since input coordinates are integers.
-    assert issubclass(m_bb.min_vertex.dtype.type, numpy.float)
-    assert issubclass(m_bb.max_vertex.dtype.type, numpy.float)
+    assert issubclass(m_bb.min_vertex.dtype.type, numpy.double)
```
[the remark this answers is not in the document]
Comment (bardkw) [reply tests/test_bounding_box.py:101]: `numpy.float` is a deprecated alias for `float`. `numpy.double` is an equivalent
https://numpy.org/doc/stable/release/1.20.0-notes.html

File: smqtk_image_io/impls/image_reader/gdal_io.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Purg) [reply smqtk_image_io/impls/image_reader/gdal_io.py:409]: Ahha! Good point. This is fine enough then. I might suggest in other cases we can use `self.__class__.__name__` to get at the name, which is useful if the class may be sub-classed so as to not have to specialize this message for each sub-class (not particularly the case here).

File: tests/utils/test_image_utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Purg) [reply tests/utils/test_image_utils.py:71]: :+1: Resolve this when you have removed it.

File: tests/test_bounding_box.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Purg) [reply tests/test_bounding_box.py:101]: :+1: Sounds good.

File: tests/impls/image_reader/test_gdal_io.py [hunk 4]
```diff
@@ -419,8 +418,9 @@ def test_load_as_matrix_empty_data(self):
         """
         empty_de = DataMemoryElement(readonly=True, content_type='image/png')
         assert empty_de.is_empty()
-        msg = "GdalImageReader cannot load 0-sized data"
-        with pytest.raises(ValueError, match=msg):
+        msg = "GdalImageReader cannot load 0-sized data (no bytes in {})." \
```
[the remark this answers is not in the document]
Comment (Purg) [reply tests/impls/image_reader/test_gdal_io.py:421]: If it wasn't clear in earlier, I meant that I would prefer minimal code changes unless you have found an active issue, in which case there should be a clarifying comment or change note.

[the remark this answers is not in the document]
Comment (bardkw) [reply tests/impls/image_reader/test_gdal_io.py]: reverted this change

**Comment by bardkw** [comment]
as for the test_file.dat, I realized the history wasnt copied with it, however i rationalized that since it was just an empty file that was previously only moved once that it would be acceptable if forgotten (same goes for the test pngs that weren't copied over until this branch)

**Pull request merged**